src/main/java/aspects/UserRestoreAspect.java

```diff
@@ -0,0 +1,63 @@
+package aspects;
+
+import auth.DjangoAuth;
+import auth.HqAuth;
+import auth.TokenAuth;
+import beans.AuthenticatedRequestBean;
+import hq.models.PostgresUser;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.integration.support.locks.LockRegistry;
+import org.springframework.web.bind.annotation.RequestMapping;
+import repo.impl.PostgresUserRepo;
+import services.RestoreFactory;
+import util.UserUtils;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+
+/**
+ * Aspect to configure the RestoreFactory
+ */
+@Aspect
+public class UserRestoreAspect {
+
+    private final Log log = LogFactory.getLog(UserRestoreAspect.class);
+
+    @Autowired
+    protected RestoreFactory restoreFactory;
+
+    @Autowired
+    protected PostgresUserRepo postgresUserRepo;
+
+    @Before(value = "@annotation(annotations.UserLock)")
+    public void configureRestoreFactory(JoinPoint joinPoint) throws Throwable {
+        Object[] args = joinPoint.getArgs();
+        if (!(args.length > 1 && args[0] instanceof AuthenticatedRequestBean && args[1] instanceof String)) {
+            throw new RuntimeException("Could not configure RestoreFactory with args " + Arrays.toString(args));
+        }
+
+        AuthenticatedRequestBean requestBean = (AuthenticatedRequestBean) args[0];
+        HqAuth auth = getAuthHeaders(requestBean.getDomain(), requestBean.getUsername(), (String) args[1]);
+        restoreFactory.configure((AuthenticatedRequestBean)args[0], auth);
+    }
+
+    private HqAuth getAuthHeaders(String domain, String username, String sessionToken) {
```

Comment:
Still used in a few places, unfortunately. Auth should also be moved into a Service